geeksurvey/views.py

```diff
@@ -109,6 +112,146 @@ def profile_update(request):
 
     return render(request, 'profile/update.html', context)
 
+@login_required
+def profile_fund(request):
+    payment = Payment(owner=request.user, amount=125)
+    payment.save()
+    request.session['payment-id'] = str(payment.id)
+
+    profile = Profile.objects.get(user=request.user)
+
+    paypal_dict = {
+            # TODO get this from decouple config
+            "business": config('PAYPAL_BIZ_EMAIL_P'),
+            "amount": payment.amount,
+            "currency_code": "USD",
+            "item_name": 'Fund GeekSurvey Account',
+            "invoice": str(payment.id),
+            "notify_url": request.build_absolute_uri(reverse('paypal-ipn')),
+            "return_url": request.build_absolute_uri(reverse('paypal-return')),
+            "cancel_return": request.build_absolute_uri(reverse('paypal-cancel')),
+            "lc": 'EN',
+            "no_shipping": '1',
+        }
+
+    form = PayPalPaymentsForm(initial=paypal_dict)
+    context = {
+        'profile': profile,
+        'form': form,
+        }
+
+    return render(request, 'profile/fund.html', context)
+
+@login_required
+def profile_claim(request):
+    # GET:
+    # claim.html contains a form where
+    # the user enter their paypal email address
+    # -
+    # POST:
+    # if valid email address, send payment to paypal
+    # from BIZ to user_addr
+    # of amount profile.balance
+    # in IPN, set their balance to 0
+    if request.method == 'GET':
+        profile = Profile.objects.get(user=request.user)
+        form    = ClaimFundsForm()
+        form['email'].label = "Enter a VALID email address for your Paypal account"
+
+        context = {
+                'form': form,
+                'profile': profile,
+                }
+
+        return render(request, 'profile/claim.html', context)
+    elif request.method == 'POST':
+
+        '''
+        Here we manually hit Paypals REST API
+        Watch out for cyber attacks!
+        -
+        To move this to a live payment system,
+            PAYOUT_URL and TOKEN_URL must be changed
+            to paypals real API instead of the
+            sandbox
+        '''
+
+        form = ClaimFundsForm(request.POST)
+        receiver = form['email'].value()
+
+        PAYOUT_URL = "https://api.sandbox.paypal.com/v1/payments/payouts"
+        TOKEN_URL = "https://api-m.sandbox.paypal.com/v1/oauth2/token"
+
+        
+        CLIENT_ID = config("PAYPAL_CLIENT_ID")
+        CLIENT_SECRET = config("PAYPAL_CLIENT_SECRET")
+
+
+        profile = Profile.objects.get(user=request.user)
+
+
+        if profile.balance < 5:
+            return redirect('profile')
+            
+        payment = Payment(owner=request.user, amount=-1*profile.balance)
```

Comment:
May want to put ```-1*profile.balance``` in parenthesis to avoid potential at-a-glance confusion with the subtraction assignment operator (```-=```).